comp/logs/agent/config/config_keys.go

```diff
@@ -110,15 +110,22 @@ func (l *LogsConfigKeys) devModeUseProto() bool {
 }
 
 func (l *LogsConfigKeys) compressionKind() string {
-	compressionKind := l.getConfig().GetString(l.getConfigKey("compression_kind"))
-	switch compressionKind {
-	case "zstd", "gzip":
-		log.Debugf("Logs agent is using: %s compression", compressionKind)
+	configKey := l.getConfigKey("compression_kind")
+	compressionKind := l.getConfig().GetString(configKey)
+
+	if l.prefix != "logs_config." {
+		pipelineCompressionKind := l.getConfig().GetString(l.prefix + "compression_kind")
+		log.Debugf("Pipeline %s compression settings - configured: %s", l.prefix, pipelineCompressionKind)
+	}
+
+	if compressionKind == "zstd" || compressionKind == "gzip" {
```

Comment:
NIT does it make sense to use the compression kind constants here/ 
```suggestion
	if compressionKind == zstdCompressionKind || compressionKind == gzipCompressionKind {
```